### ISSUE TITLE
Implementation of Mask RCNN in TensorFlow 2.x

### ISSUE THREAD

**Issue by ulhaqi12** [matterport/Mask_RCNN#2246]
Hi, Developer community,
I hope you all are in good health. I am using Mask RCNN to build an application and having problems with a version mismatch. because some other things I found online are in TensorFlow 2.x while Mask RCNN is TensorFlow 1.x. It would be great if you'll consider it to provide it in TensorFlow 2.x also.
Thanks

**Comment by saqib1707** [comment]
Hello @ulhaqi12, You can try upgrading your TF-1.x project to TF-2.x using this official tool provided by TensorFlow [here](https://www.tensorflow.org/guide/upgrade). I did the conversion using this and it works perfectly fine for me in TF-2.x

**Comment by innat** [comment]
@saqib1707 sorry, but it would not work.
@ulhaqi12 best way to downgrade your tensorflow version, (currently) `tf. 2.x` is not supported.

**Comment by saqib1707** [comment]
@innat thanks for your reply. As I mentioned above, it did work for me. Your suggestion also seems good.

**Comment by ulhaqi12** [comment]
Thanks for your suggestions, I have successfully tested Mask RCNN keras model on PC. but I am not able to convert it in tflite. There is some custom layers used in this model. I don't know how to deal with that.

**Comment by burhr2** [comment]
Hi kindly see this issue https://github.com/matterport/Mask_RCNN/issues/2312 there is a pointer to the same implementation that supports TensorFlow 2=>

**Comment by alexander-pv** [comment]
Hello, maybe it is outdated but I updated Mask-RCNN for tensorflow v2 and added some more backbones for tests including mobilenet and efficientnet architectures. For now coco weights are not included.
You can check this out: [link](https://github.com/alexander-pv/maskrcnn_tf2)

**Comment by innat** [comment]
@waleedka 
This `MaskRCNN` implementation is vastly used among the `tf. keras` practitioners still now. But unfortunately, it's been not updated for a long time. It should be updated for `tensorflo.keras 2. x`. Could you please look into this issue? **Is PR welcomed?** IMO, the limitation it has now are the a) backbones and the b) advance augmentation i.e cutmix, mixup, etc, and official support for the core API's interface update. 

@alexander-pv  
That's a really good job. I think it would be better to keep everything in one place. Have you considered PR on that? If @waleedka approves the contribution, please send PR.
